src/driver/src/workload_parsers.cpp

```diff
@@ -109,7 +157,54 @@ void WorkloadParser::convertExternal(std::string key, YAML::Node value, YAML::No
     }
 }
 
+void WorkloadParser::parseTemplates(YAML::Node templates) {
+    for (auto temp : templates) {
+        _context->insert(temp["TemplateName"].as<std::string>(), temp["Config"], Type::kActorTemplate);
+    }
+}
+
+YAML::Node WorkloadParser::parseOnlyIn(YAML::Node onlyIn) {
+    YAML::Node out;
+    YAML::Node nop;
+    nop["Nop"] = true;
+    int max = recursiveParse(onlyIn["Max"]).as<int>();
+    for (int i = 0; i <= max; i++) {
+        bool isActivePhase = false;
+        for (auto activePhase : recursiveParse(onlyIn["Active"])) {
+            if (activePhase.as<int>() == i) {
+                out.push_back(recursiveParse(onlyIn["Config"]));
+                isActivePhase = true;
+                break;
+            }
+        }
+        if (!isActivePhase) {
+           out.push_back(nop);
+        }
+    }
+    return out;
+}
+
+YAML::Node WorkloadParser::parseInstance(YAML::Node instance) {
+    YAML::Node actor;
+
+    {
+        ContextGuard guard(_context);
+        auto temp = _context->get(instance["Template"].as<std::string>(), Type::kActorTemplate);
```

Comment:
had to read this a couple times to pasre `temp` as `template` rather than `temporary`. Same applies to another place or two - can you rename to `auto template`?

Reply:
Unfortunately `template` gets interpreted as a keyword by the compiler, so I changed it to `templateNode`.